reconcile/openshift_saas_deploy_trigger_configs.py

```diff
@@ -44,29 +44,17 @@ def run(dry_run, thread_pool_size=10):
     while error:
         error = False
         for job_spec in trigger_specs:
-            saas_file_name = job_spec['saas_file_name']
-            env_name = job_spec['env_name']
-            instance_name = job_spec['instance_name']
-            job_name = get_openshift_saas_deploy_job_name(
-                saas_file_name, env_name, settings)
-            if job_name not in already_triggered:
-                logging.info(['trigger_job', instance_name, job_name])
-                if dry_run:
-                    already_triggered.append(job_name)
-
-            if not dry_run:
-                jenkins = jenkins_map[instance_name]
-                try:
-                    if job_name not in already_triggered:
-                        jenkins.trigger_job(job_name)
-                        already_triggered.append(job_name)
-                    saasherder.update_config(job_spec)
-                except Exception as e:
-                    error = True
-                    logging.error(
-                        f"could not trigger job {job_name} " +
-                        f"in {instance_name}. details: {str(e)}"
-                    )
+            trigger_options = {
+                'dry_run': dry_run,
+                'spec': job_spec,
+                'jenkins_map': jenkins_map,
+                'already_triggered': already_triggered,
+                'settings': settings,
+                'state_update_method': saasherder.update_config,
+            }
+            trigger_error = osdt_base.trigger(trigger_options)
```

Comment:
Way better:

```python
trigger_error = osdt_base.trigger(
    dry_run=dry_run,
    spec=job_spec,
    jenkins_map=jenkins_map,
    already_triggered=already_triggered
    ...
)
```
And you enjoy named arguments, instead of having Perl-in-Python. ;)
And I also believe this is better served by a `try-except` block so:

```python
try:
    osdt_base.trigger( # The stuff I put above
     )
except Exception:
     logging.exception("Problem when running osdt_base.trigger")
     time.sleep(10)
```
which also makes `osdt_base.trigger` shorter.